### DISCUSSION TITLE
How to switch mailer at runtime with dynamic values from DB?

### DISCUSSION THREAD

**Discussion by entestaweb** [adonisjs/core#4151] [Help]
I am building an application where users can add their custom SMTP.
I am storing the SMTP details in DB.
I want to send emails from different SMTP accounts at runtime.

Any feedback or suggestion will be really appreciated.

**Comment by aikrom** [comment]
Out of the box, it is not possible to use dynamic configs for existing mailer drivers, but you can create your own driver with the ability to set dynamic configs.

But I have not tested it in practice.

Links:
https://docs.adonisjs.com/guides/mailer#creating-a-custom-mail-driver
https://github.com/adonisjs/mail/blob/develop/src/Drivers/Smtp.ts
